aws-android-sdk-iot/src/main/java/com/amazonaws/mobileconnectors/iot/AWSIotProxiedSocketFactory.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ *    http://aws.amazon.com/apache2.0
+ *
+ * This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+ * OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.amazonaws.mobileconnectors.iot;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.Proxy;
+import java.net.Socket;
+
+import javax.net.ssl.SSLSocketFactory;
+
+public class AWSIotProxiedSocketFactory extends SSLSocketFactory {
+    private final SSLSocketFactory socketFactory;
+    private final Proxy proxy;
+
+    public AWSIotProxiedSocketFactory(SSLSocketFactory delegate, String proxyHost, int proxyPort){
+        this.socketFactory = delegate;
+        this.proxy = new Proxy(Proxy.Type.SOCKS, new InetSocketAddress(proxyHost, proxyPort));
```

Comment:
How did you choose SOCKS as the proxy type? Is it recommended?

Reply:
We cannot use HTTP Proxy type since we are connecting using MQTT protocol and therefore decision to choose SOCKS proxy. We will use HTTP proxy for connections using MQTT over Websocket but that is not currently supported by paho client.